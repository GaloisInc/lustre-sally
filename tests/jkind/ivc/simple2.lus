-- IVC for ok: x, y

node main(a1, a2 : int) returns ();
var
   x, y, z, ok : bool;
let
  x = y and z;
  y = a1 > 0;
  z = a2 > 10;
  assert x;

  ok = a1 > 0;
  --%IVC x, y, z;
  --%PROPERTY ok;
tel;
