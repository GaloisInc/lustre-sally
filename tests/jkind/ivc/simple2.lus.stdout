Sally Model
===========

(define-state-type
   S
   (
     (a1 Int)
     (a2 Int)
     (ok Bool)
     (x Bool)
     (y Bool)
     (z Bool)
     (|gal-initializing| Bool))
   (
     (a1 Int)
     (a2 Int)))
(define-transition-system
   TS
   S
   (= |gal-initializing| true)
   (and
      (= next.|gal-initializing| false)
      (= next.a1 input.a1)
      (= next.a2 input.a2)
      next.x
      (=
         next.y
         (< 0 next.a1))
      (=
         next.z
         (< 10 next.a2))
      (=
         next.x
         (and next.y next.z))
      (=
         next.ok
         (< 0 next.a1))))

(query
   TS
   (or |gal-initializing| ok))

[Lustre] Validating properties:
[Lustre] Property ok...
  [Sally][Valid] using 0 past states
[Lustre] Summary:
  [Valid] 1
  [Unknown] 0
  [Invalid] 0
[Lustre] Model status: [Valid]
